workflow/snakemake_rules/select_strains.smk

```diff
@@ -0,0 +1,129 @@
+'''
+
+build_dir = config.get("build_dir", "builds")
+
+def get_titers_for_build(w):
+    return "data/{lineage}/{center}_{passage}_{assay}_titers.tsv".format(**config['builds'][w.build_name])
+
+rule titer_priorities:
+    input:
+        titers = get_titers_for_build,
+        metadata = lambda w: f"data/{config['builds'][w.build_name]['lineage']}/metadata.tsv"
+    output:
+        priorities = build_dir + "/{build_name}/titer_priorities.tsv"
+    run:
+        import pandas as pd
+        metadata = pd.read_csv(input.metadata, sep='\t')
+        titer_counts = {s:0 for s in metadata.strain}
+
+        titers = pd.read_csv(input.titers, sep='\t')
+        for ri, row in titers.iterrows():
+            test_strain = row.iloc[0]
+            if test_strain in titer_counts:
+                titer_counts[test_strain] += 1
+
+        with open(output.priorities, 'w') as fh:
+            for s,p in titer_counts.items():
+                fh.write(f"{s}\t{p}\n")
+
+def get_subsample_input(w):
+    files = {"metadata": f"data/{config['builds'][w.build_name]['lineage']}/metadata.tsv"}
+    if config['builds'][w.build_name]['subsamples'][w.subsample].get('priorities', '')=='titers':
+        files['titers']=build_dir + f"/{w.build_name}/titer_priorities.tsv"
+    return files
+
+rule subsample:
+    input:
+        unpack(get_subsample_input)
+    output:
+        strains = build_dir + "/{build_name}/strains_{subsample}.txt",
+    params:
+        filters =  lambda w: config["builds"][w.build_name]["subsamples"][w.subsample]["filters"],
+        priorities = lambda w: f"--priority {build_dir}/{w.build_name}/titer_priorities.tsv" \
+                               if config['builds'][w.build_name]['subsamples'][w.subsample].get('priorities', '')=='titers' else ''
+    conda: "environment.yaml"
+    shell:
+        """
+        augur filter \
+            --metadata {input.metadata} \
+            {params.filters} \
+            {params.priorities} \
+            --output-strains {output.strains} 2>&1 | tee {log}
+        """
+
+rule select_strains:
+    input:
+        metadata = lambda w: f"data/{config['builds'][w.build_name]['lineage']}/metadata.tsv",
+        subsamples = lambda w: [f"{build_dir}/{w.build_name}/strains_{s}.txt" for s in config['builds'][w.build_name]['subsamples']]
+    output:
+        strains = build_dir + "/{build_name}/strains.txt",
+    run:
+        strains = set()
+        for fname in input.subsamples:
+            with open(fname) as fh:
+                for line in fh:
+                    strains.add(line.strip())
+
+        with open(output.strains, 'w') as fh:
+            fh.write('\n'.join(strains)+'\n')
+
+rule select_sequences:
+    input:
+        strains = build_dir + "/{build_name}/strains.txt",
+        sequences = lambda w: f"data/{config['builds'][w.build_name]['lineage']}/{w.segment}.fasta"
+    output:
+        sequences = build_dir + "/{build_name}/{segment}/sequences.fasta"
+    shell:
```

Comment:
We can do all of this with a single augur filter command. I don’t think our flu data are so big that we need to use seqkit and custom rules to concat strain lists. The `select_sequences` rule should just be a call to augur filter with `--exclude-all --include <strain list files> --output-sequences sequences.fasta --output-metadata metadata.tsv`. It will be fast enough and allow us to drop a bunch of separate rules.